lib/elixir/lib/kernel/typespec.ex

```diff
@@ -174,11 +174,9 @@ defmodule Kernel.Typespec do
   @spec defines_spec?(module, atom, arity) :: boolean
   def defines_spec?(module, name, arity)
       when is_atom(module) and is_atom(name) and arity in 0..255 do
-    finder = fn {_kind, expr, _caller} ->
+    Enum.any?(Module.get_attribute(module, :spec), fn {_kind, expr, _caller} ->
       spec_to_signature(expr) == {name, arity}
-    end
-
-    :lists.any(finder, Module.get_attribute(module, :spec))
+    end)
```

Comment:
I was able to do this here and in `defines_callback?/3` below but I ran into bootstrapping problems doing this in https://github.com/elixir-lang/elixir/pull/7437/files#diff-6de4b356bc9d4e3d07f1e5731fbe7d23L167 (`defines_type?/3`), where it complained that `Enum.any?/2` was undefined 🤔

Reply:
defines_type? is used by Protocols and we need protocols to compile Enum. So I guess we can't do it for defines_type? although that's probably fine.

Reply:
Ah I see, cool!